### PR TITLE
Clean last address and scann all addresses

### LINKED ISSUE #2871: When changing BWS setting, update address indexes in new BWS
It could happen that the new BWS has 'old' indexes,
so money transfered to cached address will be in a 'future' address.

Also, the address cache new to be cleaned with BWS setting is changed.

### PR DESCRIPTION
**Pull request by cmgustavo**
fixes #2871

### CODE REVIEW

File: src/js/controllers/index.js [hunk 1]
```diff
@@ -549,9 +548,8 @@ angular.module('copayApp.controllers').controller('indexController', function($r
     c.startScan({
       includeCopayerBranches: true,
     }, function(err) {
-      if (err) {
-        if (self.walletId == walletId)
-          self.setOngoingProcess('scanning', false);
+      if (err && self.walletId == walletId) {
```
Comment (matiu) [inline src/js/controllers/index.js]: Not sure about this change. If an error occur and the wallet is focused it should be shown.

File: src/js/controllers/index.js [hunk 2]
```diff
@@ -624,6 +622,22 @@ angular.module('copayApp.controllers').controller('indexController', function($r
 
   $rootScope.$on('Local/BWSUpdated', function(event) {
     profileService.applyConfig();
+    storageService.setCleanAndScanAddresses(function() {});
+  });
+
+  $rootScope.$on('Local/CleanAndScanAddresses', function(event) {
+    storageService.getCleanAndScanAddresses(function(err, val) {
```
Comment (matiu) [inline src/js/controllers/index.js]: I think this should not be an other event, but it should be in 'newfocusedwallet'

File: src/js/controllers/index.js (hunk 1 above)
Comment (cmgustavo) [inline src/js/controllers/index.js]: That is exactly what does. If wallet is focused, display an error. If don't, not. For example, if we are trying to scan a wallet (not focused) that doesn't exist on the new BWS, it displays the message to recreate in the *focused* wallet.

File: src/js/controllers/index.js (hunk 2 above)
Comment (cmgustavo) [inline src/js/controllers/index.js]: Good idea!

**Comment by cmgustavo** [comment]
Updated.

File: src/js/controllers/index.js [hunk 3]
```diff
@@ -769,6 +768,19 @@ angular.module('copayApp.controllers').controller('indexController', function($r
     self.setFocusedWallet();
     self.updateTxHistory();
     go.walletHome();
+    storageService.getCleanAndScanAddresses(function(err, val) {
+      if (val) {
+        $log.debug('Clear last address cache and Scan');
+        var wallets = profileService.walletClients;
+        for (var walletId in wallets) {
```
Comment (matiu) [inline src/js/controllers/index.js]: lodash.each(lodash.keys(profileService.walletClients),  function(walletId) {
   storageSe...
});

**Pull request merged**